lib/resty/core/base64.lua

```diff
@@ -42,12 +43,25 @@ ngx.encode_base64 = function (s)
             s = tostring(s)
         end
     end
+
     local slen = #s
-    local dlen = base64_encoded_length(slen)
-    -- print("dlen: ", tonumber(dlen))
-    local dst = get_string_buf(dlen)
-    dlen = C.ngx_http_lua_ffi_encode_base64(s, slen, dst)
-    return ffi_string(dst, dlen)
+    if no_padding then
+        if no_padding ~= true then
+            return error("boolean argument only")
+        end
+
+        local dlen = base64_encoded_length(slen, true)
+        local dst = get_string_buf(dlen)
+        local r_dlen = C.ngx_http_lua_ffi_encode_base64(s, slen, dst, 1)
+        -- if dlen ~= r_dlen then ngx.say ("discrepancy in len") end
+        return ffi_string(dst, r_dlen)
+    else
+        local dlen = base64_encoded_length(slen, false)
+        local dst = get_string_buf(dlen)
+        local r_dlen = C.ngx_http_lua_ffi_encode_base64(s, slen, dst, 0)
+        -- if dlen ~= r_dlen then ngx.say ("discrepancy in len") end
+        return ffi_string(dst, r_dlen)
```

Comment:
I think we can avoid the code duplication here? I don't see why we need a branch here in the first place ;) We can just pass down the `no_padding` argument to those low level functions like `base64_encoded_length` and `C.ngx_http_lua_ffi_encode_base64`, right?